tests/ui/needless_return.rs

```diff
@@ -461,3 +461,42 @@ pub unsafe fn issue_12157() -> *const i32 {
     return unsafe { todo() } as *const i32;
     //~^ needless_return
 }
+
+mod else_ifs {
+    fn test1(a: i32) -> u32 {
+        if a == 0 {
+            return 1;
+        //~^ needless_return
+        } else if a < 10 {
+            return 2;
+        //~^ needless_return
+        } else {
+            return 3;
+            //~^ needless_return
+        }
+    }
+
+    fn test2(a: i32) -> u32 {
+        if a == 0 {
+            return 1;
+        //~^ needless_return
+        } else if a < 10 {
+            2
+        } else {
+            return 3;
+            //~^ needless_return
+        }
+    }
+
+    fn test3(a: i32) -> u32 {
+        if a == 0 {
+            return 1;
+        //~^ needless_return
+        } else if a < 10 {
+            2
+        } else {
+            return 3;
+            //~^ needless_return
+        }
+    }
```

Comment:
Could you add some weirded expressions, for future proofing (and checking that it works correctly right now)

```suggestion
    fn test4(a: i32) -> u32 {
    	if a == 0 {
    		return 1;
    	} else if if if a > 0x1_1 {
    			return 2;
    	} else {
    		return 5;
    	} {true} else { true } { 0xDEADC0DE }
    	else if match a { b @ _ => { return 1; } }
    	{ 0xDEADBEEF }
    	else { 1 } 
    }
    }
```